producer-services/src/main/scala/za/co/absa/spline/producer/service/repo/ExecutionProducerRepositoryImpl.scala

```diff
@@ -137,77 +138,74 @@ class ExecutionProducerRepositoryImpl @Autowired()(db: ArangoDatabaseAsync) exte
 }
 
 object ExecutionProducerRepositoryImpl {
+
+  import SimpleJsonSerDe._
+
   private[repo] def createEventKey(e: ExecutionEvent) =
     s"${e.planId}:${jl.Long.toString(e.timestamp, 36)}"
 
-  private def createExecutes(executionPlan: ExecutionPlan) = EdgeDef.Executes.edge(
+  private def createExecutes(executionPlan: model.ExecutionPlan) = EdgeDef.Executes.edge(
     executionPlan.id,
     s"${executionPlan.id}:${executionPlan.operations.write.id}")
 
-  private def createExecution(executionPlan: ExecutionPlan): Execution = {
-    val extras = executionPlan
-      .extraInfo
-      .updated("systemInfo", executionPlan.systemInfo)
-      .optionally[AgentInfo](_.updated("agentInfo", _), executionPlan.agentInfo)
-
-    Execution(
-      extra = extras,
+  private def createExecution(executionPlan: model.ExecutionPlan): dbModel.ExecutionPlan =
+    dbModel.ExecutionPlan(
+      systemInfo = executionPlan.systemInfo.toJsonAs[Map[String, Any]],
+      agentInfo = executionPlan.agentInfo.map(_.toJsonAs[Map[String, Any]]).orNull,
+      extra = executionPlan.extraInfo,
       _key = executionPlan.id.toString)
-  }
 
-  private def createReadsFrom(plan: ExecutionPlan, dsUriToKey: String => String): Seq[Edge] = for {
+  private def createReadsFrom(plan: model.ExecutionPlan, dsUriToKey: String => String): Seq[Edge] = for {
     ro <- plan.operations.reads
     ds <- ro.inputSources
   } yield EdgeDef.ReadsFrom.edge(
     s"${plan.id}:${ro.id}",
     dsUriToKey(ds))
 
-  private def createWriteTo(executionPlan: ExecutionPlan, dsUriToKey: String => String) = EdgeDef.WritesTo.edge(
+  private def createWriteTo(executionPlan: model.ExecutionPlan, dsUriToKey: String => String) = EdgeDef.WritesTo.edge(
     s"${executionPlan.id}:${executionPlan.operations.write.id}",
     dsUriToKey(executionPlan.operations.write.outputSource))
 
-  private def createExecutionDepends(plan: ExecutionPlan, dsUriToKey: String => String): Seq[Edge] = for {
+  private def createExecutionDepends(plan: model.ExecutionPlan, dsUriToKey: String => String): Seq[Edge] = for {
     ro <- plan.operations.reads
     ds <- ro.inputSources
   } yield EdgeDef.Depends.edge(
     plan.id,
     dsUriToKey(ds))
 
-  private def createExecutionAffects(executionPlan: ExecutionPlan, dsUriToKey: String => String) = EdgeDef.Affects.edge(
+  private def createExecutionAffects(executionPlan: model.ExecutionPlan, dsUriToKey: String => String) = EdgeDef.Affects.edge(
     executionPlan.id,
     dsUriToKey(executionPlan.operations.write.outputSource))
 
   private def createDataSources(dsUriToKey: Map[String, String]): Seq[DataSource] = dsUriToKey
     .map({ case (uri, key) => DataSource(uri, key) })
     .toVector
 
-  private def createOperations(executionPlan: ExecutionPlan): Seq[Operation] = executionPlan.operations.all.map {
+  private def createOperations(executionPlan: model.ExecutionPlan): Seq[dbModel.Operation] = executionPlan.operations.all.map {
```

Comment:
```suggestion
  private def createOperations(executionPlan: rest.ExecutionPlan): Seq[db.Operation] = executionPlan.operations.all.map {
```
This would be easiest for me to read. 

`restModel` and `dbModel` would be ok as well though longer than needed I think.

Reply:
I named it `restModel` originally, but then realized that it's not actually specific to REST API and can be used for other API's as well. That's why I renamed `producer-rest-model` to `producer-model`. What other name can we use instead of `model` in this case? `serviceModel`?

Reply:
What about `apiModel`?

Reply:
For me it's more about to which layer of application things belong and I have rest associated most with the layer higher than db. But `apiModel` seems ok as well. There's probably no 100% solution.

Reply:
Ok let's do `apiModel` then.